src/azure-cli/azure/cli/command_modules/acr/_utils.py

```diff
@@ -469,3 +453,125 @@ def prepare_source_location(cmd, source_location, client_registries, registry_na
+    actions = []
+    for rule in gateway_actions_list:
+        gateway = rule[0]
+        if len(rule) < 2:
+            raise CLIError('At least one action must be specified with the gateway {}.'.format(gateway))
+        for action in rule[1:]:
+            action = action.lower()
+            if action not in valid_actions:
+                raise CLIError('Invalid action "{}" provided. \nValid actions are {}.'.format(action, valid_actions))
+            actions.append('{}/{}/{}'.format('gateway', gateway, action))
+
+    return actions
+
+
+def create_default_scope_map(cmd,
+                             resource_group_name,
+                             registry_name,
+                             scope_map_name,
+                             repositories,
+                             gateways,
+                             scope_map_description="",
+                             force=False):
+    from ._client_factory import cf_acr_scope_maps
+    scope_map_client = cf_acr_scope_maps(cmd.cli_ctx)
+    actions = parse_actions_from_repositories(repositories)
+    actions.extend(parse_actions_from_gateways(gateways))
+    try:
+        existing_scope_map = scope_map_client.get(resource_group_name, registry_name, scope_map_name)
+        # for command idempotency, if the actions are the same, we accept it
+        if sorted(existing_scope_map.actions) == sorted(actions):
+            return existing_scope_map
+        if force and existing_scope_map:
+            logger.warning("Overriding scope map '%s' properties", scope_map_name)
+        else:
+            raise CLIError('The default scope map was already configured with different repository permissions.' +
+                           '\nPlease use "az acr scope-map update -r {} -n {} --add <REPO> --remove <REPO>" to update.'
+                           .format(registry_name, scope_map_name))
+    except CloudError:
+        pass
+    logger.warning('Creating a scope map "%s" for provided permissions.', scope_map_name)
+    poller = scope_map_client.create(resource_group_name, registry_name, scope_map_name,
+                                     actions, scope_map_description)
+    scope_map = LongRunningOperation(cmd.cli_ctx)(poller)
+    return scope_map
+
+
+def build_token_id(subscription_id, resource_group_name, registry_name, token_name):
+    return "/subscriptions/{}/resourceGroups/{}".format(subscription_id, resource_group_name) + \
+            "/providers/Microsoft.ContainerRegistry/registries/{}/tokens/{}".format(registry_name, token_name)
+
+
+def get_token_from_id(cmd, token_id):
+    from ._client_factory import cf_acr_tokens
+    from .token import acr_token_show
+    token_client = cf_acr_tokens(cmd.cli_ctx)
+    # SCOPE MAP ID example
+    #/subscriptions/<1>/resourceGroups/<3>/providers/Microsoft.ContainerRegistry/registries/<7>/tokens/<9>'
+    token_info = token_id.lstrip('/').split('/')
+    if len(token_info) != 10:
```

Comment:
Is it a common pattern in cli to parse the resource?

Reply:
We don't do this anywhere else in ACR but I'm not sure if other teams do it or not.
Optionally I could do a request to get the token to verify it exists and get the ID from the response